bam_ampliconclip.c

```diff
@@ -526,6 +528,49 @@ static inline void swap_bams(bam1_t **a, bam1_t **b) {
 }
 
 
+// Format OA:Z:(RNAME,POS,strand,CIGAR,MAPQ,NM;
+static inline int tag_original_data(bam1_t *orig, kstring_t *oa_tag) {
+    char strand;
+    uint8_t *nm_tag, *old_oa_tag;
+    uint32_t *cigar;
+    int64_t nm = 0;
+    int i;
+
+    ks_clear(oa_tag);
+
+    if ((old_oa_tag = bam_aux_get(orig, "OA"))) {
+        ksprintf(oa_tag, "%s", bam_aux2Z(old_oa_tag));
```

Comment:
Strictly, there should be error checking on these kstring functions.

Reply:
It is not something we normally do, not in samtools, not in htslib.

Reply:
I'm trying to encourage it, e.g. b2d1b9ff.  Ignoring kstring function return values can lead to silent failures so it's best to check them.

Reply:
You can of course do:
```
int e = 0;
e |= ksprintf(...) < 0;
e |= kputw(...) < 0;
// ... etc
if (e) return -1;
```
to reduce the number of branches needed.